### PR TITLE
Add "Check external command version for compatibility"

### LINKED ISSUE #286: add "Check external command version for compatibility" example
Check if version of external command (i.e. parse output of "$my_binnary --version" external command, or output of some http request to a versioned API)
for compatibility with your application's requirements (i.e. our shellout would work only with given version range of ffmpeg tool). 

Use the [semver](https://github.com/steveklabnik/semver) crate

cc https://github.com/rust-lang-nursery/rust-cookbook/issues/276

### PR DESCRIPTION
**Pull request by bobbo**
fixes #286

### CODE REVIEW

File: src/app.md [hunk 1]
```diff
@@ -725,13 +726,63 @@ fn run() -> Result<()> {
...
+    let version_test = VersionReq::parse("> 1.12.0")?;
+    let output = Command::new("git").arg("--version").output()?;
+
+    if !output.status.success() {
+        bail!("Command executed with failing error code");
+    }
+
+    let stdout = String::from_utf8(output.stdout)?;
+    // `git --version` output: "git version x.y.z"
+    let version = stdout.split(" ").last().expect("Invalid command output");
```
Comment (budziq) [inline src/app.md]: lets avoid using `expect` here. you can add custom error message with `ok_or_else(|| "Invalid command ...")?`

File: src/app.md [hunk 2]
```diff
@@ -725,13 +726,63 @@ fn run() -> Result<()> {
...
+    let parsed_version = Version::parse(version)?;
+
+    if !version_test.matches(&parsed_version) {
+        bail!("Command version lower than minimum supported version")
```
Comment (budziq) [inline src/app.md]: how about adding the parsed version number to the `bail!` invocation? 
`bail!` has format capabilities identical to the `println!`

**Comment by bobbo** [comment]
Hi @budziq, I've made the changes you suggested

**Comment by budziq** [comment]
Well done! Thanks @bobbo !

**Pull request merged**